hs_core/views/utils.py

```diff
@@ -1027,74 +1027,73 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
     zip_with_full_path = os.path.join(resource.root_path, zip_with_rel_path)
-
     if not resource.supports_unzip(zip_with_rel_path):
         raise ValidationError("Unzipping of this file is not supported.")
 
     unzip_path = None
     try:
-        if overwrite:
-            # unzip to a temporary folder
-            unzip_path = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
-            # list all files to be moved into the resource
-            unzipped_files = listfiles_recursively(istorage, unzip_path)
-            unzipped_foldername = os.path.basename(unzip_path)
-
-            res_files = []
-            for unzipped_file in unzipped_files:
-                res_files.append(IrodsFile(unzipped_file, istorage))
-            meta_files = []
-            if ingest_metadata:
-                res_files, meta_files, map_files = identify_metadata_files(res_files)
-            # walk through each unzipped file, delete aggregations if they exist
-            for file in res_files:
-                destination_file = _get_destination_filename(file.name, unzipped_foldername)
-                if (istorage.exists(destination_file)):
-                    if resource.resource_type == "CompositeResource":
-                        aggregation_object = resource.get_file_aggregation_object(
-                            destination_file)
-                        if aggregation_object:
-                            aggregation_object.logical_delete(user)
-                        else:
-                            logger.error("No aggregation object found for " + destination_file)
-                            istorage.delete(destination_file)
+        # unzip to a temporary folder
+        unzip_path = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
+        # list all files to be moved into the resource
+        unzipped_files = listfiles_recursively(istorage, unzip_path)
+        unzipped_foldername = os.path.basename(unzip_path)
+        override_tgt_paths = []
+
+        res_files = []
+        for unzipped_file in unzipped_files:
+            res_files.append(IrodsFile(unzipped_file, istorage))
+        meta_files = []
+        if ingest_metadata:
+            res_files, meta_files, map_files = identify_metadata_files(res_files)
+
+        for file in res_files:
+            destination_file = _get_destination_filename(file.name, unzipped_foldername)
+            if istorage.exists(destination_file):
+                override_tgt_paths.append(destination_file)
+
+        if not overwrite and override_tgt_paths:
+            override_simplified_paths = [tgt_path.split('data/contents/')[1] for tgt_path in override_tgt_paths]
+            raise FileOverrideException('move would overwrite {}'.format(', '.join(override_simplified_paths)))
+
+        for override_tgt_path in override_tgt_paths:
+            if (istorage.exists(override_tgt_path)):
+                if resource.resource_type == "CompositeResource":
```

Comment:
I think we can remove this check for composite resource as well.  Everything will have been migrated to composite by Aug 1.